modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/core/utils/sectionAddedHandler.js

```diff
@@ -79,13 +79,27 @@ const handleSectionAdded = (props, state, event) => {
 
 	let newPages;
 
-	if (parentFieldName) {
+	if (existingField.type === FIELD_TYPE_FIELDSET) {
+		newPages = addFieldToPage({
+			...props,
+			indexes: {
+				columnIndex: 0,
+				pageIndex: 0,
+				rowIndex: existingField.rows.length,
```

Comment:
Hey, @carolmariaabb. Is it possible for `existingField.rows` to be undefined?

Reply:
@rodrigopaulino I did some tests and I couldn’t find a way. I think this case would be a bug because it's not possible to create a `fieldset` without a field inside.
But I can change to `existingField.rows?.length || 0` just to be sure, what do you think?

Reply:
I'm fine with leaving it like this and I agree with your findings. I just wanted to be sure it was intentional. 😊